test/Fields/reduction_cuda.jl

```diff
@@ -229,3 +229,68 @@ end
     @test LinearAlgebra.norm(Yf, 3) ≈ LinearAlgebra.norm(Yf_cpu, 3)
     @test LinearAlgebra.norm(Yf, Inf) ≈ LinearAlgebra.norm(Yf_cpu, Inf)
 end
+
+@testset "test cuda reduction op for single column" begin
+    FT = Float64
+    context = ClimaComms.SingletonCommsContext(ClimaComms.CUDADevice())
+    context_cpu = ClimaComms.SingletonCommsContext(ClimaComms.CPUDevice()) # CPU context for comparison
+
+    z0 = FT(0)
+    z_max = FT(2 * π)
+    z_nelems = 16
+
+    println(
+        "running column reduction test on $(context.device); z_nelems = $z_nelems; z0 = $z0; z_max = $z_max; FT = $FT",
+    )
+
+    domain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(z0),
+        Geometry.ZPoint{FT}(z_max);
+        boundary_tags = (:left, :right),
+    )
+
+    z_mesh = Meshes.IntervalMesh(domain; nelems = z_nelems)
+    z_topology = Topologies.IntervalTopology(context, z_mesh)
+    z_topology_cpu = Topologies.IntervalTopology(context_cpu, z_mesh)
+
+
+    center_space = Spaces.CenterFiniteDifferenceSpace(z_topology)
+    face_space = Spaces.FaceFiniteDifferenceSpace(center_space)
+
+    center_space_cpu = Spaces.CenterFiniteDifferenceSpace(z_topology_cpu)
+    face_space_cpu = Spaces.FaceFiniteDifferenceSpace(center_space_cpu)
+
+    Yc = ones(FT, center_space)
+    Yc_cpu = ones(FT, center_space_cpu)
+
+    Yf = ones(FT, face_space)
+    Yf_cpu = ones(FT, face_space_cpu)
```

Comment:
In the future, can we move some of this boiler plate into helper functions? Whenever I go to fix performance issues, I often have to hoist everything outside of `@testset` and put it into a function. It would also help de-boat our test.